recipes/openmpi/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class OpenMPIConan(ConanFile):
+    name = "openmpi"
+    homepage = "https://www.open-mpi.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "mpi", "openmpi")
+    description = "A High Performance Message Passing Library"
+    license = "https://www.open-mpi.org/community/license.php"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "fortran": ["yes", "mpifh", "usempi", "usempi80", "no"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "fortran": "no"
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("OpenMPI doesn't support Windows")
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
```

Comment:
openmpi has also a vendored `libevent` (and maybe others libs). Don't know if there is a configure option to use external one.

Reply:
yes you can use external libevent: https://github.com/open-mpi/ompi/blob/master/config/opal_config_libevent.m4

Reply:
seems to be possible:
```
  --with-libevent=DIR     Search for libevent headers and libraries in DIR.
                          Should only be used if an external copy of libevent
                          is being used.
  --with-libevent-libdir=DIR
                          Search for libevent libraries in DIR. Should only be
                          used if an external copy of libevent is being used.
```
I'll try it

Reply:
weird, it didn't work:
```
checking --with-libevent-libdir value... not found
configure: WARNING: Expected file /Users/sse4/.conan/data/libevent/2.1.12/_/_/package/7ac42c52e63f1381a4948b0f2939b106f29c4ea2/lib/libevent.* not found
```
the files libevent has:
```
➜  all git:(openmpi) ✗ ls /Users/sse4/.conan/data/libevent/2.1.12/_/_/package/7ac42c52e63f1381a4948b0f2939b106f29c4ea2/lib/
libevent_core.a     libevent_extra.a    libevent_openssl.a  libevent_pthreads.a
```

Reply:
@SSE I suggest to add a FIXME to unvendor what we can in another PR, and move forward with this one.

Reply:
added